File: backend-java/src/main/java/com/skewwhiffy/auraltester/notation/model/abc/ClefNotes.java
package com.skewwhiffy.auraltester.notation.model.abc;

import com.skewwhiffy.auraltester.notation.model.note.NoteSequence;

public interface ClefNotes extends NoteSequence {
    /*

interface ClefNotes : NoteSequence {
  val AbsoluteNote.downOne: AbsoluteNote
    get() = (this - Interval.minor(2)).ignoreAccidental

  val AbsoluteNote.upOne: AbsoluteNote
    get() = (this + Interval.minor(2)).ignoreAccidental

  val AbsoluteNote.skipOne: AbsoluteNote
    get() = (this + Interval.minor(3)).ignoreAccidental

  val AbsoluteNote.ignoreAccidental: AbsoluteNote
    get() = Note(note.noteName, Accidental.natural)
      .let { AbsoluteNote(it, octave, lyric) }

  val AbsoluteNote.withNoteName: AbsoluteNote
    get() = withLyric(note.displayString)
}
     */
}

File: backend/src/main/kotlin/com/skewwhiffy/notation/model/abc/ClefNotes.kt
package com.skewwhiffy.notation.model.abc

import com.skewwhiffy.notation.model.*

interface ClefNotes : NoteSequence {
  val AbsoluteNote.downOne: AbsoluteNote
    get() = (this - Interval.minor(2)).ignoreAccidental

  val AbsoluteNote.upOne: AbsoluteNote
    get() = (this + Interval.minor(2)).ignoreAccidental

  val AbsoluteNote.skipOne: AbsoluteNote
    get() = (this + Interval.minor(3)).ignoreAccidental

  val AbsoluteNote.ignoreAccidental: AbsoluteNote
    get() = Note(note.noteName, Accidental.natural)
      .let { AbsoluteNote(it, octave, lyric) }

  val AbsoluteNote.withNoteName: AbsoluteNote
    get() = withLyric(note.displayString)
}